lib/yaml.ml

```diff
@@ -183,3 +189,16 @@ let pp ppf s =
   match to_string s with
   | Ok s -> Format.pp_print_string ppf s
   | Error (`Msg m) -> Format.pp_print_string ppf (Printf.sprintf "(error (%s))" m)
+
+let rec equal v1 v2 =
+  match v1, v2 with
+  | `Null, `Null -> true
+  | `Bool x1, `Bool x2 -> ((=) : bool -> bool -> bool) x1 x2
```

Comment:
any reason not to write `x1 = x2` here?

Reply:
Oh sorry, to specialise it -- `Bool x1` is a poly variant here so it wouldnt specialise based on the types of the pattern match

Reply:
Yeah, I haven't tested it, but I think specialising helps here. We could also use `[@@deriving eq]` to replace this implementation, but that would add an extra dependency.